chalicelib/checks/wfr_checks.py

```diff
@@ -1,1448 +1,1456 @@
+    action.output["error"] = error
     return action
 
 
 @check_function()
-def SV_metawfrs_to_patch_sample_processing(connection, **kwargs):
-    """Find SV metaworkflowruns that may need sample_processing patched with processed files
+def find_meta_workflow_runs_requiring_output_linktos(connection, **kwargs):
+    """Find completed MetaWorkflowRuns to PATCH output files to desired
+    locations.
     """
-    check = CheckResult(connection, 'SV_metawfrs_to_patch_sample_processing')
-    my_auth = connection.ff_keys
-    check.action = "patch_SV_pfs_to_sample_processing"
-    check.description = "Find SV metaworkflow runs that may need sample processing to be patched."
-    check.brief_output = []
-    check.summary = ""
-    check.full_output = {}
-    check.status = 'PASS'
-
-    # check indexing queue
-    env = connection.ff_env
-    indexing_queue = ff_utils.stuff_in_queues(env, check_secondary=True)
-
-    if indexing_queue:
-        check.status = 'PASS'  # maybe use warn?
-        check.brief_output = ['Waiting for indexing queue to clear']
-        check.summary = 'Waiting for indexing queue to clear'
-        check.full_output = {}
-        return check
-
-    # start with cases with a metawfr and no ingested final vcf
-    query = '/search/?type=Case&meta_workflow_run_sv!=No+value&sv_vcf_file=No+value'
-    search_res = ff_utils.search_metadata(query, key=my_auth)
-
-    # filter those whose samples do not have processed_files
-    filtered_res = []
-    for r in search_res:
-        SNV_processed = 0
-        SV_processed = 0
-        result_list = r['sample_processing'].get('processed_files', [])
-        for pf in result_list:
-            try:
-                pf['variant_type']
-                if pf['variant_type'] == "SV":
-                    SV_processed += 1
-                elif pf['variant_type'] == "SNV":
-                    SNV_processed += 1
-            except:
-                SNV_processed += 1
-        if SV_processed < 2:
-            filtered_res.append(r)
-
-    # nothing to run
-    if not filtered_res:
-        check.summary = 'All Good!'
-        return check
-
-    metawfr_uuids = [r['meta_workflow_run_sv']['uuid'] for r in filtered_res]
-    metawfr_titles = [r['meta_workflow_run_sv']['display_title'] for r in filtered_res]
-
-    check.allow_action = True
-    check.summary = 'Some metawfrs may need patching sample processing.'
-    check.status = 'WARN'
-    msg = str(len(metawfr_uuids)) + ' metawfrs may need patching sample processing'
+    check = initialize_check(connection)
+    check.description = "Find completed MetaWorkflowRuns to PATCH their output files."
+    check.action = "link_meta_workflow_run_output_files"
+
+    meta_workflow_runs = []
+    query = (
+        "search/?type=MetaWorkflowRun&final_status=completed&field=uuid"
+        "&output_files_linked_status=No+value"
+    )
+    search_response = ff_utils.search_metadata(query, key=connection.ff_keys)
```

Comment:
Noted that you have eliminated the `stuff_in_queues` check. Just want to be sure that this is desired and you think it is safe, as it is in theory possible to get inconsistent results if edits occur at just the wrong time. This may (or may not) be okay... Generally if you care only about top level fields it is sufficient to use `stuff_in_queues` without the secondary flag, which should be a much lighter check. 

On this I tend to be of the opinion that we can accept that for some (long running) operations we can be eventually consistent. So I am okay with removing the check if you are convinced it is safe.

Reply:
As discussed via DM, this shouldn't be a problem, and I updated Magma functions to ensure I'm not operating on stale ES results containing items that were deleted.